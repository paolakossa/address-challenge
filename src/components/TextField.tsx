import React from "react";
import { useController } from "react-hook-form";

type Props = {
  name: string;
  label?: string;
  valueDefault?: string;
  placeholder?: string;
  type?: string;
};

const TextField = ({
  name,
  label,
  valueDefault = "",
  placeholder,
  type,
  ...props
}: Props) => {
  const {
    field,
    fieldState: { invalid, error },
  } = useController({ name, defaultValue: valueDefault });

  return (
    <div>
      <label htmlFor={name} className="block mb-1 font-medium">
        {label}
      </label>
      <div className="flex items-center">
        <input
          id={name}
          {...field}
          {...props}
          type={type}
          placeholder={placeholder}
          className="border p-2 rounded flex-1"
        />
      </div>

      {invalid && <span>{error?.message}</span>}
    </div>
  );
};

export default TextField;
