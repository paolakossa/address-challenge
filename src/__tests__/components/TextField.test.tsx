import { render, screen } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';
import TextField from '../../components/TextField';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Wrapper = ({ children }: any) => {
  const methods = useForm();
  return <FormProvider {...methods}>{children}</FormProvider>;
};

describe('TextField', () => {
  it('renders correctly with label', () => {
    render(
      <Wrapper>
        <TextField name="testInput" label="Test Input" />
      </Wrapper>
    );

    expect(screen.getByLabelText('Test Input')).toBeInTheDocument();
  });

  it('renders correctly without label', () => {
    render(
      <Wrapper>
        <TextField name="testInput" />
      </Wrapper>
    );

    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Wrapper>
        <TextField name="testInput" label="Test Input" />
      </Wrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});




