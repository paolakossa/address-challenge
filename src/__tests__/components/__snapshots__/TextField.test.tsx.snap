// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block mb-1 font-medium"
      for="testInput"
    >
      Test Input
    </label>
    <div
      class="flex items-center"
    >
      <input
        class="border p-2 rounded flex-1"
        id="testInput"
        name="testInput"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
